File: src/dataset/python/difference-of-squares_exercism-ThomasZumsteg/openAI/ChatGPT4_difference_of_squares_v2.py
def square_of_sum(num):
    """Square of sum of numbers less than or equal to [num]"""
    sum_num = num * (num + 1) // 2
    return sum_num * sum_num

def sum_of_squares(num):
    """Sum of squares of numbers less than or equal to [num]"""
    return num * (num + 1) * (2 * num + 1) // 6

def difference(num):
    """Difference between the square_of_sum and sum_of_squares"""
    sum_num = num * (num + 1) // 2
    return sum_num * sum_num - num * (num + 1) * (2 * num + 1) // 6

File: src/dataset/javascript/difference-of-squares_exercism-ThomasZumsteg/openAI/ChatGPT4_difference-of-squares_v2.js
class Squares {
  constructor(num) {
    const n = num;
    const nPlus1 = n + 1;
    const nPlus1Squared = nPlus1 * nPlus1;
    this.squareOfSums = (n * n * nPlus1Squared) / 4;
    this.sumOfSquares = (n * nPlus1 * (2 * n + 1)) / 6;
    this.difference = this.squareOfSums - this.sumOfSquares;
  }
}

module.exports = Squares;